### PR TITLE
feat: add mysql check constraints

### PR DESCRIPTION
**Pull request by vmuresanu**
Adds support for creating mysql check constraints on entities with @Check decorator.
MySQL supports check constraints from v8.0.16 

Closes -  https://github.com/typeorm/typeorm/issues/4148

### CODE REVIEW

**Comment by imnotjames** [comment]
Tests aren't passing.  Can you rebase and fix the tests?

File: src/driver/mysql/MysqlQueryRunner.ts [hunk 1]
```diff
@@ -949,28 +957,54 @@ export class MysqlQueryRunner extends BaseQueryRunner implements QueryRunner {
...
+        const promises = checkConstraints.map(checkConstraint => this.createCheckConstraint(tableOrName, checkConstraint));
+        await Promise.all(promises);
     }
 
     /**
      * Drops check constraint.
      */
     async dropCheckConstraint(tableOrName: Table|string, checkOrName: TableCheck|string): Promise<void> {
-        throw new Error(`MySql does not support check constraints.`);
+        const table = tableOrName instanceof Table ? tableOrName : await this.getCachedTable(tableOrName);
```
Comment (imnotjames) [inline src/driver/mysql/MysqlQueryRunner.ts:988]: Needs version check

File: src/driver/mysql/MysqlQueryRunner.ts [hunk 2]
```diff
@@ -1689,6 +1730,15 @@ export class MysqlQueryRunner extends BaseQueryRunner implements QueryRunner {
         return c;
     }
 
+    protected createCheckConstraintSql(table: Table, checkConstraint: TableCheck): Query {
+        return new Query(`ALTER TABLE ${this.escapePath(table)} ADD CONSTRAINT ${checkConstraint.name} CHECK (${checkConstraint.expression})`);
```
Comment (imnotjames) [inline src/driver/mysql/MysqlQueryRunner.ts:1734]: Should the constraint name be quoted?

**Review by imnotjames — CHANGES_REQUESTED** [review]
Please rebase and fix tests

**Comment by ezzabuzaid** [comment]
Any updates...

File: src/driver/mysql/MysqlQueryRunner.ts (hunk 1 above)
Comment (vmuresanu) [reply src/driver/mysql/MysqlQueryRunner.ts:1025]: Fixed the comments, thank you very much. And also tests written by me passed. But unfortunately some random tests fail and to be honest I didn't find how to fix them. Can you please give me a hit on how to fix them?

Comment (gabrielsaruhashi) [reply src/driver/mysql/MysqlQueryRunner.ts:1025]: @imnotjames bumping vmuresanu's request!
This feature addition would be greatly appreciated :)

**Comment by imnotjames** [comment]
Tests are still failing.

**Comment by imnotjames** [comment]
Please feel free to add tests & reopen a new PR.  This looks like an interesting feature.

**Pull request closed without merging**